db/migrate/20181121194330_create_messages.rb

```diff
@@ -0,0 +1,12 @@
+class CreateMessages < ActiveRecord::Migration[5.2]
+  def change
+    create_table :messages do |t|
+      t.text  :body
+      t.boolean :is_read, default: false
+      t.references :messageable, polymorphic: true
+      t.references :user
+      t.timestamps
+    end
+    add_index :messages, [ :user_id, :is_read ]
```

Comment:
what about indexes by messagable and user_id?

Reply:
t.references :user add index too
# == Schema Information
#
# Table name: messages
# Indexes
#
#  index_messages_on_messageable_type_and_messageable_id  (messageable_type,messageable_id)
#  index_messages_on_user_id                              (user_id)
#  index_messages_on_user_id_and_is_read                  (user_id,is_read)
#